### DISCUSSION TITLE
I want to implement it on webpage

### DISCUSSION THREAD

**Discussion by jarvisx17** [juanmc2005/diart#155] [General]
Is there any way i can implement socket on webpage using javascript or javascript socket or python socket like diart.client.

**Comment by juanmc2005** [comment]
Hi @jarvisx17, this is currently out of the scope of the library. Diart focuses on providing streaming and real-time speaker diarization (and soon transcription starting with v0.8). It aims to provide the tools so that you can build web, mobile, desktop, or other types of applications.

The cli interface of `diart.serve` and `diart.client` are not meant to be production-ready, but rather quick prototyping tools in order to test your own diart servers and clients.

In other words, there is no plan to implement javascript clients in the library.

**Reply by juanmc2005** [reply]
That said, it is indeed possible to create a websocket client in javascript and send audio chunks to a server running `diart.serve` for example.

**Reply by jarvisx17** [reply]
Yes i have created the javascript client but the results are not accurate. I've use the base64 encoding.

Do you have any idea how we can got good accuracy with javascript client or if you can suggest me the steps to develop it.

**Reply by juanmc2005** [reply]
If you haven't already, I suggest you follow the exact same encoding as in `diart.utils.encode_audio()` in your client:

```python
def encode_audio(waveform: np.ndarray) -> Text:
    data = waveform.astype(np.float32).tobytes()
    return base64.b64encode(data).decode("utf-8")
```

I don't know what the equivalent javascript function would look like.
I'm also open to changing this encoding/decoding algorithm if you happen to know a more efficient and/or practical one.
